Revert this upstream commit...

  https://chromium.googlesource.com/chromium/src/+/e33287758f2234d6aabfc5d4e011c4e81e3a47cf
  [rust] Remove the old __rust_alloc_error_handler_should_panic symbol

...which leads to link errors like the following with Rust < 1.90 (?):

  FAILED: proto_extras_plugin
  [...]
  ld.lld-19: error: undefined symbol: __rust_alloc_error_handler_should_panic
  >>> referenced by alloc.rs:435 (library/alloc/src/alloc.rs:435)
  >>>               alloc-58205e8b81e6ba4b.alloc.429db6835e310c2b-cgu.5.rcgu.o:(__rdl_oom) in archive prebuilt_rustc_sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc.rlib
  >>> referenced by alloc.rs:354 (library/std/src/alloc.rs:354)
  >>>               std-aca21d74f3a8f2a2.std.231170a795f74ec8-cgu.14.rcgu.o:(std::alloc::default_alloc_error_hook::hc0ab80c3377fb5a3) in archive prebuilt_rustc_sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd.rlib
  clang++-19: error: linker command failed with exit code 1 (use -v to see invocation)


--- a/build/rust/allocator/lib.rs
+++ b/build/rust/allocator/lib.rs
@@ -105,6 +105,13 @@
     static __rust_no_alloc_shim_is_unstable: u8 = 0;
 
     // Mangle the symbol name as rustc expects.
+    // TODO(crbug.com/440481922): Remove this after rolling past https://github.com/rust-lang/rust/pull/143387
+    #[rustc_std_internal_symbol]
+    #[allow(non_upper_case_globals)]
+    #[linkage = "weak"]
+    static __rust_alloc_error_handler_should_panic: u8 = 0;
+
+    // Mangle the symbol name as rustc expects.
     #[rustc_std_internal_symbol]
     #[allow(non_upper_case_globals)]
     #[linkage = "weak"]
